Fixes lava turfs processing forever as well as some burn logic

MrsTonedOne opened the pull request:
fixes #14260
closes #14271 (my pr is better then this pr)

MrsTonedOne committed:
Fixes lava turfs processing forever as well as some burn logic

optimumtact commented:
:+1:

KorPhaeron commented:
Thanks

RemieRichards commented on code/game/turfs/simulated/floor/plating.dm:
```diff
@@ -193,30 +193,32 @@
 
 /turf/simulated/floor/plating/lava/proc/burn_stuff()
+	. = 0
 	for(var/atom/movable/AM in contents)
```
istypeless-for please.

Aranclanos replied:
do this

MrStonedOne replied:
Can't, there could be non-movables in the contents.

RemieRichards replied:
Yes MSO, which is why you do ```/atom```, the AM part is unused, since all code is under ```istype(obj)``` or ```istype(mob/living)```

And this won't increase the number of objects looped through either, since byond already loops through them all, it just has to ```istype()``` them which is unneeded here

MrStonedOne replied:
point.

MrsTonedOne committed:
does the remie

MrsTonedOne committed:
blame RR

[23:27:53] \<%Remie\> you know how it makes inflammable things burn
[23:28:02] \<%Remie\> (and that's buggy as fug but w/e)
[23:28:10] \<%Remie\> what about resetting it after the fire_act() call?
[23:28:21] \<%Remie\> that way inflammable things become inflammable again if they escape the lava
